File: code/widgets/org.eclipsescout.demo.widgets.client/src/org/eclipsescout/demo/widgets/client/ui/desktop/outlines/WidgetsOutline.java
/*******************************************************************************
 * Copyright (c) 2015 BSI Business Systems Integration AG.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Distribution License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/org/documents/edl-v10.html
 * 
 * Contributors:
 *     BSI Business Systems Integration AG - initial API and implementation
 ******************************************************************************/
package org.eclipsescout.demo.widgets.client.ui.desktop.outlines;

import java.util.Collection;

import org.eclipse.scout.commons.exception.ProcessingException;
import org.eclipse.scout.rt.client.ui.desktop.outline.pages.IPage;
import org.eclipse.scout.rt.extension.client.ui.desktop.outline.AbstractExtensibleOutline;
import org.eclipse.scout.rt.shared.TEXTS;
import org.eclipsescout.demo.widgets.client.ui.desktop.outlines.pages.FormFieldsNodePage;
import org.eclipsescout.demo.widgets.client.ui.desktop.outlines.pages.MenusNodePage;

public class WidgetsOutline extends AbstractExtensibleOutline {

  @Override
  protected String getConfiguredTitle() {
    return TEXTS.get("WidgetsOutline");
  }

  @Override
  protected void execCreateChildPages(Collection<IPage> pageList) throws ProcessingException {
    FormFieldsNodePage formFieldNodePage = new FormFieldsNodePage();
    pageList.add(formFieldNodePage);
    pageList.add(new MenusNodePage());
  }
}

File: code/widgets/org.eclipse.scout.widgets.ui.html/src/main/js/widgets/WidgetsOutline.js
/*
 * Copyright (c) 2014-2019 BSI Business Systems Integration AG.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     BSI Business Systems Integration AG - initial API and implementation
 */
widgets.WidgetsOutline = function() {
  widgets.WidgetsOutline.parent.call(this);
};
scout.inherits(widgets.WidgetsOutline, scout.Outline);

widgets.WidgetsOutline.prototype._createOutlineOverview = function() {
  return scout.create('widgets.WidgetsTileOutlineOverview', {
    parent: this,
    outline: this
  });
};
